ui/src/utils/url.ts

```diff
@@ -1 +1,32 @@
-export const urlWithProtocol = (url: string) => `https://${url.replace('https://', '')}`;
+// Logic adapted from:
+// https://github.com/distribution/distribution/blob/7b502560cad43970472964166dcb095b1f883ae4/reference/normalize.go
+
+const protocol = 'https://';
+const dockerHub = 'hub.docker.com';
+const officialPath = '_';
+const userPath = 'r';
+
+export const urlForImage = (image: string): string => {
+  const parts = image.split('/');
+
+  if (parts[0] === 'docker.io') {
+    parts.shift();
+  }
+
+  if (parts[0] === 'library') {
+    parts.shift();
+  }
+
+  image = parts.join('/');
+
+  if (parts.length === 1) {
+    return `${protocol}${dockerHub}/${officialPath}/${image}`;
+  } else if (
```

Comment:
I have another one:

public.ecr.aws/karpenter/controller:v0.31.2
can be seen at:
https://gallery.ecr.aws/karpenter/controller

Reply:
Actually, no-one can easily publish to that repository so no point in adding it.

Reply:
And here is another:

602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/coredns

will map to:

https://us-west-2.console.aws.amazon.com/ecr/repositories/private/602401143452/eks/coredns

Reply:
Added support and a test 👍🏻